### ISSUE TITLE
Implement weapons (no hitscan code or explosion code, just weapon raise/lower/pickup)

### ISSUE THREAD

**Issue by WChrisK** [Helion-Engine/Helion#45]
This is simple implementation of weapons. We should be able to raise/lower them. This has no relation to rendering, that will be handled in another issue. Same for hitscans or missiles that explode.

Depends on #44
